src/sinks/splunk_hec.rs

```diff
@@ -117,13 +117,11 @@ impl HttpSink for HecSinkConfig {
 
         let host = event.get(&self.host_key).cloned();
 
-        let timestamp = if let Some(Value::Timestamp(ts)) =
-            event.remove(&event::log_schema().timestamp_key())
-        {
-            ts.timestamp_nanos()
-        } else {
-            chrono::Utc::now().timestamp_nanos()
+        let timestamp = match event.remove(&event::log_schema().timestamp_key()) {
+            Some(Value::Timestamp(ts)) => ts,
+            _ => chrono::Utc::now(),
         };
+        let timestamp = (timestamp.timestamp_millis() as f64) / 1000f64;
```

Comment:
This is definitely incorrect. Floats can't accurately represent whole number domains. (Eg 1.00001, 1.000002, 1.000003, ...).

While many test runs won't see this, I'm fairly sure this is the reson why `humio_valid_time_field` test fails fairly often on CI.